pkgs/development/libraries/gtk+/3.x.nix

```diff
@@ -21,12 +21,14 @@ stdenv.mkDerivation rec {
     sha256 = "1l45nd7ln2pnrf99vdki3l7an5wrzkbak11hnnj1w6r3fkm4xmv1";
   };
 
+  NIX_LDFLAGS = stdenv.lib.optionalString stdenv.isDarwin "-lintl";
```

Comment:
Yes, that's right. See https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/default.nix#L141 and https://github.com/NixOS/nix/blob/master/src/libexpr/primops.cc#L488-L491